amazon_transcribe/serialize.py

```diff
@@ -25,62 +24,62 @@
     BaseEvent,
 )
 
-HEADER_VALUE = Union[int, None, str]
 
-
-class Serializer:
-    def __init__(self):
-        raise NotImplementedError("Serializer")
-
-    def serialize(self) -> Tuple[Dict[str, HEADER_VALUE], BufferedIOBase]:
-        """Serialize out to payload and headers."""
-        raise NotImplementedError("serialize")
-
-    def serialize_to_request(self) -> PreparedRequest:
-        """Serialize parameters into an HTTP request."""
-        raise NotImplementedError("serialize_to_request")
-
-
-class TranscribeStreamingRequestSerializer(Serializer):
+class TranscribeStreamingSerializer:
     """Convert StartStreamTranscriptionRequest into a
     Request object for streaming to the Transcribe service.
     """
 
-    def __init__(self, endpoint, transcribe_request):
-        self.endpoint: str = endpoint
-        self.method: str = "POST"
-        self.request_uri: str = "/stream-transcription"
-        self.request_shape: StartStreamTranscriptionRequest = transcribe_request
+    def serialize_start_stream_transcription_request(
+        self, endpoint: str, request_shape: StartStreamTranscriptionRequest
+    ) -> Request:
 
-    def serialize(self) -> Tuple[Dict[str, HEADER_VALUE], BufferedIOBase]:
-        headers = {
-            "x-amzn-transcribe-language-code": self.request_shape.language_code,
-            "x-amzn-transcribe-sample-rate": self.request_shape.media_sample_rate_hz,
-            "x-amzn-transcribe-media-encoding": self.request_shape.media_encoding,
-            "x-amzn-transcribe-vocabulary-name": self.request_shape.vocabulary_name,
-            "x-amzn-transcribe-session-id": self.request_shape.session_id,
-            "x-amzn-transcribe-vocabulary-filter-method": self.request_shape.vocab_filter_method,
-            "x-amzn-transcribe-vocabulary-filter-name": self.request_shape.vocab_filter_name,
-            "x-amzn-transcribe-show-speaker-label": self.request_shape.show_speaker_label,
-            "x-amzn-transcribe-enable-channel-identification": self.request_shape.enable_channel_identification,
-            "x-amzn-transcribe-number-of-channels": self.request_shape.number_of_channels,
-        }
+        method: str = "POST"
+        request_uri: str = "/stream-transcription"
 
-        _add_required_headers(self.endpoint, headers)
+        headers: Dict[str, str] = {}
+        headers["x-amzn-transcribe-language-code"] = str(request_shape.language_code)
```

Comment:
Rather than just casting everything to a string, what do you think about doing a bit more robust serialization set up.

We can create some general serialization functions like `_serialize_int_header`, `_serialize_bool_header`, etc. These can take a header name and value, do some rough validation on the typing and whatever specific mutations we need to do into a string. The end goal would be a code generator now has a specific serialize X header and can just plug that into the output code with a header name and a value.